accessories/button.coffee

```diff
@@ -8,17 +8,32 @@ module.exports = (env) ->
 
   class ButtonAccessory extends DefaultAccessory
 
-    constructor: (device) ->
-      super(device, Service.Switch)
+    constructor: (device, buttonToTrigger) ->
+      # this handling is needed in order to support pimatic devices
+      # which need to be represented by mutiple homekit devices
+      # as function overloading is not supportet in node
+      if !buttonToTrigger?
+        # buttonToTrigger was omitted in constructor call thus we need to call super
+        # without additional parameters to propagate that they were not manipulated
+        # basically this is legacy handling to support previous behaviour with
+        # support for only one button
+        super(device, Service.Switch)
+        button = device.config.buttons[0]
+      else
+        #manipulate device name to reflect buttons-device + button name
+        deviceName = "#{device.name} #{buttonToTrigger.text}"
+        #manipulate device ID to allow multiple instances of device
+        deviceId = device.id + buttonToTrigger.id
```

Comment:
would be good to separate the ids with e.g. underscore, so you can separate the ids.